plugins/net_plugin/net_plugin.cpp

```diff
@@ -2633,6 +2628,12 @@ namespace eosio {
          fc::optional<transaction_id_type> trx_id;
          fc::raw::unpack( ds, trx_id );
          if( trx_id ) {
+            if (trx_in_progress_sz > def_max_trx_in_progress_size) {
```

Comment:
needs the exact same block of code before line 2649, except you will replace *trx_id with ptr->id().  This will make sure that we check the max trx when the optional trx_id is not present.  This would be similar to the trx_message_v0_which case, where we have to read in the complete transaction to be able to calculate the transaction's id.